### ISSUE TITLE
[Docs]: About multinode setup

### ISSUE THREAD

**Issue by wangkuiyi** [sgl-project/sglang-jax#518]
### Checklist

- [x] 1. Make sure you already searched for relevant issues and documentations.
- [x] 2. Please use English, otherwise it will be closed.

### Documentation Issues or Suggested Changes

https://github.com/sgl-project/sglang-jax/blob/main/docs/architecture/project-core-structure.md includes a figure, but not specifically for the multinode setup. After reading the code, here is my mental model -- a service hosted on 4 TPU VMs, each with multiple TPUs. Could the team please check if my understanding is correct? Thanks!

```
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                    SGLang-JAX Multi-Node Architecture (4 TPU Hosts)                                                                      │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

     NODE 0 (Coordinator)                    NODE 1 (Worker)                      NODE 2 (Worker)                      NODE 3 (Worker)
┌─────────────────────────────┐        ┌─────────────────────────────┐      ┌─────────────────────────────┐      ┌─────────────────────────────┐
│                             │        │                             │      │                             │      │                             │
│  ┌───────────────────────┐  │        │  ┌───────────────────────┐  │      │  ┌───────────────────────┐  │      │  ┌───────────────────────┐  │
│  │   Main Process        │  │        │  │   Main Process        │  │      │  │   Main Process        │  │      │  │   Main Process        │  │
│  │   HTTP + Tokenizer    │◄─┼──┐     │  │   (health check only) │  │      │  │   (health check only) │  │      │  │   (health check only) │  │
│  └───────────┬───────────┘  │  │     │  └───────────────────────┘  │      │  └───────────────────────┘  │      │  └───────────────────────┘  │
│              │ ZMQ          │  │     │                             │      │                             │      │                             │
│              ▼              │  │     │                             │      │                             │      │                             │
│  ┌───────────────────────┐  │  │pub  │  ┌───────────────────────┐  │ pub  │  ┌───────────────────────┐  │ pub  │  ┌───────────────────────┐  │
│  │   Scheduler           │──┼──┼sub──┼─►│   Scheduler           │◄─┼─sub──┼─►│   Scheduler           │◄─┼─sub──┼─►│   Scheduler           │  │
│  │   [TPU]               │◄─┼──┼jax──┼─►│   [TPU]               │◄─┼─jax──┼─►│   [TPU]               │◄─┼─jax──┼─►│   [TPU]               │  │
│  └───────────┬───────────┘  │  │dist │  └───────────────────────┘  │ dist │  └───────────────────────┘  │ dist │  └───────────────────────┘  │
│              │ ZMQ          │  │     │                             │      │                             │      │                             │
│              ▼              │  │     │                             │      │                             │      │                             │
│  ┌───────────────────────┐  │  │     │                             │      │                             │      │                             │
│  │   Detokenizer         │──┼──┘     │                             │      │                             │      │                             │
│  └───────────────────────┘  │ ZMQ    │                             │      │                             │      │                             │
│                             │        │                             │      │                             │      │                             │
│  3 processes                │        │  2 processes                │      │  2 processes                │      │  2 processes                │
└─────────────────────────────┘        └─────────────────────────────┘      └─────────────────────────────┘      └─────────────────────────────┘

Communication:
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
• ZMQ IPC (ipc://)   : Node 0 internal communication (Tokenizer → Scheduler → Detokenizer → Tokenizer)
• ZMQ TCP (pub/sub)  : Node 0 Scheduler broadcasts requests to Nodes 1,2,3 Schedulers
• jax.distributed    : SPMD coordination across all TPU chips (collective ops like all-reduce)

TOTAL: 9 processes (3 + 2 + 2 + 2)
